src/validation_utils.py

```diff
@@ -55,3 +97,46 @@ def check_column_clusters(cols: dict, data_frame: pd.DataFrame):
                 if not pd.isna(row).all():
                     result.append((index, items))
     return result
+
+def unique_values(df: pd.DataFrame, col_name: str):
+    """Check if all values in the column are unique."""
+    return df[col_name].nunique() == len(df[col_name])
+
+def find_none_values(df: pd.DataFrame, col_name: str):
+    """Check if all values in the column contains None."""
+    return(df[df[col_name].isnull()].index.tolist())
+
+def identifier_checks(data: dict, id_col_names: list) -> bool:
+    """Check identifier columns."""
+
+    # check if id columns contain None values
+    for col_name in id_col_names:
+        for data_name in [d for d in data if col_name in data[d]]:
+            res = find_none_values(data[data_name], col_name)
+            if len(res) > 0:
+                print(f"{FAIL}{data_name} column {col_name} empty cells found in rows\n{res}{ENDC}")
+                print("Check no empty cells in ID columns completed.")
+                return False
+
+    # check if ids in host.csv and environment.csv are unique 
+    if not unique_values(data["hosts.csv"], "host_id"):
+        print(f"{FAIL}hosts.csv: Column host_id contains duplicates.{ENDC}")
+        return False
+    if not unique_values(data["environment.csv"], "environment_id"):
+        print(f"{FAIL}environment.csv: Column environment_id contains duplicates.{ENDC}")
+        return False
+    
+    #TODO: generalise the tuple (file, id_col)
+    ids = {}
+    ids["host_id"] = data["hosts.csv"]["host_id"].unique()
+    ids["environment_id"] = data["environment.csv"]["environment_id"].unique()
+    
+    for col_name in id_col_names:
+        for data_name in [d for d in data if col_name in data[d]]:
+            if not set(data[data_name][col_name]).issubset(ids[col_name]):
+                print(f"{FAIL}File {data_name} contains undefined ids in column {col_name}:")
+                print(set(data[data_name][col_name]).difference(ids[col_name]))
+                print(ENDC)
+                #return False
```

Comment:
I guess this is a warning now? Should we be consistent with a warning message instead then?

Reply:
I need to uncomment the return False .... If something is wrong with the identifier columns, we should not continue the checking.